app/browser/api/ledger.js

```diff
@@ -2173,44 +2025,17 @@ const onCallback = (state, result, delayTime) => {
   }
 
   if (notInProgress) {
-    state = module.exports.cacheRuleSet(state, regularResults.ruleset)
-    if (result.has('rulesetV2')) {
-      results = regularResults.rulesetV2 // TODO optimize if possible
-      delete regularResults.rulesetV2
-
-      entries = []
-      results.forEach((entry) => {
-        const key = entry.facet + ':' + entry.publisher
-
-        if (entry.exclude !== false) {
-          entries.push({type: 'put', key: key, value: JSON.stringify(underscore.omit(entry, ['facet', 'publisher']))})
-        } else {
-          entries.push({type: 'del', key: key})
-        }
-      })
-
-      v2RulesetDB.batch(entries, (err) => {
-        if (err) return console.error(v2RulesetPath + ' error: ' + JSON.stringify(err, null, 2))
-
-        if (entries.length === 0) return
-
-        const publishers = ledgerState.getPublishers(state)
-        for (let item of publishers) {
-          const publisherKey = item[0]
-          const publisher = item[1] || Immutable.Map()
-
-          if (!publisher.getIn(['options', 'exclude'])) {
-            excludeP(publisherKey, (unused, exclude) => {
-              appActions.onPublisherOptionUpdate(publisherKey, 'exclude', exclude)
-              savePublisherOption(publisherKey, 'exclude', exclude)
-            })
-          }
-        }
-      })
-    }
+    const publishers = ledgerState.getPublishers(state)
+    for (let item of publishers) {
+      const publisherKey = item[0]
```

Comment:
Aren't publishers immutable object?

Reply:
I will double check, but this was basically preserved from the previous `if (notInProgress)` block here: https://github.com/brave/browser-laptop/pull/14751/files#diff-3cdfdb124f718787d584e9e2b82fcedcL2198

Reply:
@NejcZdovc you can iterate over a `Map` using `for (let item of map)`

Reply:
Yeah I was too fast with this one, I though that you were doing something else, sorry